vortex-array/src/arrow/compute/to_arrow/temporal.rs

```diff
@@ -112,3 +112,24 @@ where
 
     Ok(Arc::new(ArrowPrimitiveArray::<T>::new(values, nulls)))
 }
+
+fn to_arrow_timestamp<T: ArrowTimestampType>(
+    array: &TemporalArray,
+    arrow_tz: &Option<Arc<str>>,
+) -> VortexResult<ArrowArrayRef>
+where
+    T::Native: NativePType,
+{
```

Comment:
This shouldn't be a completely separate function, change `to_arrow_temporal` to accept timezone instead?

Reply:
Done, although I had to use a shared function. `to_arrow_temporal` can't do anything with a timezone since it is generic over `ArrowPrimitiveType` and `with_timezone` is only available with `ArrowTimestampType`s. I felt like casting to remove code duplication was the wrong tradeoff. The other option would be to remove `to_arrow_temporal` and call `to_arrow_temporal_inner` in all the callsites, but that requires some boilerplate (maybe there's a macro somewhere I can use, can also write). Happy to do whatever you prefer.